lib/Backends/CPU/CPUFunction.h

```diff
@@ -27,21 +27,35 @@ class CPUFunction final : public CompiledFunction {
   /// The LLVM JIT engine. The jit must be initialized after the ctor
   /// initializes the LLVM backends.
   std::unique_ptr<llvm::orc::GlowJIT> JIT_;
-  /// This represents the heap, that stores the activations at runtime.
-  void *heap_;
+  /// runtimeBundle contains symbol offsets and allocation sizes.
+  runtime::RuntimeBundle runtimeBundle_;
+  /// Base address for Activations memory block.
+  uint8_t *baseActivationsAddress_;
+  /// Base address for Mutable weights memory block, Inputs and Outputs.
+  uint8_t *baseMutableWeightVarsAddress_;
 
 public:
   /// Ctor.
-  CPUFunction(std::unique_ptr<llvm::orc::GlowJIT> JIT, void *heap);
-
+  CPUFunction(std::unique_ptr<llvm::orc::GlowJIT> JIT,
+              const runtime::RuntimeBundle &runtimeBundle);
+  /// Copy Function to device, an empty function for CPU.
+  void copyFunctionToDevice(){};
+  /// Copy Constants to device, an empty function for CPU.
+  void copyConstantsToDevice(){};
+  /// Allocate Mutable buffers on device, Activations and Placeholders.
+  void allocateMutableBuffersOnDevice();
+  /// Copy Mutable data to device.
```

Comment:
what's actually "mutable data"?

It's clear what's constants, activations, placeholders.

Reply:
I don't know if we currently have a good way to filter this from the context. For this PR I suggest we copy all. I've created a separate issue #1953 to track how we'd like to pass inputs/outputs going forward.